pkg/terraform/errors/errors.go

```diff
@@ -16,40 +16,156 @@ limitations under the License.
 
 package errors
 
+import (
+	"fmt"
+	"strings"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/pkg/errors"
+)
+
+const (
+	levelError = "error"
+)
+
+type tfError struct {
+	logs    []byte
+	tfLogs  []*TerraformLog
+	cause   error
+	message string
+}
+
 type applyFailed struct {
-	log string
+	*tfError
+}
+
+// TerraformLog represents relevant fields of a Terraform CLI JSON-formatted log line
+type TerraformLog struct {
+	Level      string        `json:"@level"`
+	Message    string        `json:"@message"`
+	Diagnostic LogDiagnostic `json:"diagnostic"`
+}
+
+// LogDiagnostic represents relevant fields of a Terraform CLI JSON-formatted
+// log line diagnostic info
+type LogDiagnostic struct {
+	Severity string `json:"severity"`
+	Summary  string `json:"summary"`
+	Detail   string `json:"detail"`
+}
+
+func (t *tfError) Error() string {
+	if t == nil {
+		return ""
+	}
+
+	messages := make([]string, 0, len(t.tfLogs))
+	for _, l := range t.tfLogs {
+		// only use error logs
+		if l == nil || l.Level != levelError {
+			continue
+		}
+		m := l.Message
+		if l.Diagnostic.Severity == levelError && l.Diagnostic.Summary != "" {
```

Comment:
Do you think keeping any information other than what we're printing here has benefits over-weighing the additional logic complexity? I can see us saving a single error string in `tfError` filtered with these conditions in `parseTerraformLogs` and just print that when `Error()` is called, i.e. `applyFailed` having only `message string` field with the message that contains everything we'd like to print.

Reply:
I had preferred this so that computation of the actual (filtered) error message is deferred until `Error()` is called. But not a big deal in my opinion.

Reply:
Yeah I think the computations are negligible that the earlier and simpler we can get away, the better.